src/analyze/analyze-security.c

```diff
@@ -2380,12 +2382,105 @@ static int get_security_info(Unit *u, ExecContext *c, CGroupContext *g, Security
         return 0;
 }
 
-int analyze_security(sd_bus *bus, char **units, AnalyzeSecurityFlags flags) {
+static int offline_security_check(Unit *u) {
+        _cleanup_(table_unrefp) Table *overview_table = NULL;
+        AnalyzeSecurityFlags flags = 0;
+        _cleanup_(security_info_freep) SecurityInfo *info = NULL;
+        int r = 0, k = 0;
+
+        assert(u);
+
+        if (DEBUG_LOGGING)
+                unit_dump(u, stdout, "\t");
+
+        r = get_security_info(u, unit_get_exec_context(u), unit_get_cgroup_context(u), &info);
+        if (r < 0)
+              return r;
+        else {
+                k = assess(info, overview_table, flags);
```

Comment:
I meant here as well - we use an auxiliary return var when we don't want to lose an earlier error code if we carry on despite it. But it's not the case here - hence you can replace this whole `else` block with just `return assess(info, overview_table, flags);`